`test_device_host_file_step_by_step` test failing

It seems that `dask_cuda/tests/test_cudf_builtin_spilling.py::test_device_host_file_step_by_step` recently began to fail:

```python
==================================== FAILURES =====================================
_______________________ test_device_host_file_step_by_step ________________________

tmp_path = PosixPath('/tmp/pytest-of-pentschev/pytest-175/test_device_host_file_step_by_0')
manager = <SpillManager spill_on_demand=False device_memory_limit=N/A | 0B spilled | 24B (0%) unspilled (unspillable)>

    def test_device_host_file_step_by_step(tmp_path, manager: SpillManager):
        tmpdir = tmp_path / "storage"
        tmpdir.mkdir()
        pdf = pandas.DataFrame({"a": [1, 2, 3]})
        cdf = cudf.DataFrame({"a": [1, 2, 3]})
        dhf = DeviceHostFile(
            device_memory_limit=safe_sizeof(pdf),
            memory_limit=safe_sizeof(pdf),
            worker_local_directory=tmpdir,
        )
        dhf["pa1"] = pdf
        dhf["cu1"] = cdf

        assert set(dhf.others.keys()) == set(["cu1"])
        assert set(dhf.device.keys()) == set()
>       assert set(dhf.host.keys()) == set(["pa1"])
E       AssertionError: assert set() == {'pa1'}
E         Extra items in the right set:
E         'pa1'
E         Full diff:
E         - {'pa1'}
E         + set()

dask_cuda/tests/test_cudf_builtin_spilling.py:90: AssertionError
================================ warnings summary =================================
dask_cuda/tests/test_cudf_builtin_spilling.py::test_device_host_file_step_by_step
  /datasets/pentschev/miniconda3/envs/rn-230103/lib/python3.9/site-packages/cudf/core/buffer/spill_manager.py:428: UserWarning: overwriting non-empty manager: (<SpillableBuffer size=24B spillable=True exposed=False num-spill-locks=0 ptr=0x7f383dc00200 owner=<rmm._lib.device_buffer.DeviceBuffer object at 0x7f3b9e64ddc0>>,)
    warnings.warn(f"overwriting non-empty manager: {buffers}")

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================= short test summary info =============================
FAILED dask_cuda/tests/test_cudf_builtin_spilling.py::test_device_host_file_step_by_step - AssertionError: assert set() == {'pa1'}
===================== 1 failed, 4 passed, 1 warning in 10.11s =====================
```

I can reproduce this locally (seen above) and https://github.com/rapidsai/dask-cuda/pull/1069 is also failing on that test.

@madsbk are you aware of any changes that could have caused this?

--- a/dask_cuda/tests/test_cudf_builtin_spilling.py
+++ b/dask_cuda/tests/test_cudf_builtin_spilling.py
@@ -77,6 +77,11 @@ def test_device_host_file_step_by_step(tmp_path, manager: SpillManager):
     tmpdir.mkdir()
     pdf = pandas.DataFrame({"a": [1, 2, 3]})
     cdf = cudf.DataFrame({"a": [1, 2, 3]})
+
+    # Pandas will cache the result of probing this attribute.
+    # We trigger it here, to get consistent results from `safe_sizeof()`
+    hasattr(pdf, "__cuda_array_interface__")
+
     dhf = DeviceHostFile(
         device_memory_limit=safe_sizeof(pdf),
         memory_limit=safe_sizeof(pdf),

